lib/update/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Device independent helpers to do image update
+#![no_std]
+
+pub use drv_update_api::UpdateError;
+use userlib::TaskId;
+
+// Size of processing for the driver. A driver may pass less than this
+// size but it cannot pass greater than this size.
+pub fn block_size(update_task: TaskId) -> Result<usize, UpdateError> {
```

Comment:
I'm a little confused why this module needs to exist.  It seems like `block_size` and `start_update` forward stuff straight to the `struct Update` in `update-api`.

Looking at the Hiffy calls below, is the advantage slightly shorter code, e.g.
```rust
    func_err(libupdate::start_update(UPDATE.get_task_id()))?;
```
vs
```rust
    func_err(drv_update_api::Update::from(UPDATE.get_task_id()).start_update())?;
```
?

`write_block` does a little more work, but I wonder if it could be added as an extra `impl` block on the (autogenerated) `struct Update` in `update-api/lib.rs`, e.g.
```rust
impl Update {
    pub fn write_block(start_block: usize, data: &[u8]) -> Result<(), UpdateError> {
        let block_size = update.block_size()?;

        for (i, c) in data.chunks(block_size).enumerate() {
            update.write_one_block(start_block + i, c)?;
        }

        Ok(())
    }
}
```

Reply:
I think you're right. The extra module is a bit of a historical artifact as I was writing this the wrong way.